src/ParallelAlgorithms/Events/ObserveErrorNorms.hpp

```diff
@@ -121,8 +121,17 @@ class ObserveErrorNorms<ObservationValueTag, SubtractAnalyticSolution,
   using observed_reduction_data_tags =
       observers::make_reduction_data_tags<tmpl::list<ReductionData>>;
 
-  using argument_tags = tmpl::list<ObservationValueTag, Tensors...,
-                                   ::Tags::AnalyticSolutionsBase>;
+  // If SubtractAnalyticSolutions == true, then the final argument tag must
+  // be ::Tags::AnalyticSolutionsBase, so that the analytic solution can
+  // be retrieved and subtracted. Otherwise, this tag is not needed or used.
+  // But to avoid having to implement operator() with a variable number of
+  // argument tags, when SubtractAnalyticSolutions == false, here just
+  // pass in a tag from the DataBox, ObservationValueTag, for the final
+  // argument tag.
```

Comment:
this is a _little_ bit of a hack. You could define a helper function with the contents of the current `operator()` and overload the `operator()` with the two possible sets of arguments, each dispatching to the helper.
I'm alright with the code as-is, and writing it the way with a helper function will definitely be more verbose, I just wanted to point out the alternative.